.github/workflows/build.yml

```diff
@@ -1,32 +0,0 @@
-name: Build
-on:
-  push:
-    branches:
-      - main
-      - feat/*
-  pull_request:
-    types: [opened, synchronize, reopened]
-jobs:
-  sonarcloud:
-    # skip this on forks or dependabot PRs
-    name: SonarCloud
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-        with:
-          fetch-depth: 0 # Shallow clones should be disabled for a better relevancy of analysis
-          lfs: true
-      - name: Checkout LFS objects
-        run: git lfs checkout
-      - name: Setup Go
-        uses: actions/setup-go@v4
-        with:
-          go-version: "1.20" # The Go version to download (if necessary) and use.
-      - name: Test with coverage
-        run: go test -coverpkg=./x/... -coverprofile=coverage.out ./...
-      - name: SonarCloud Scan
```

Comment:
Should we not still have the Sonar Cloud scan? I don't see this in the workflow outputs anymore

Reply:
These are not the same thing afaict, see a screenshot of the details page for the SonarCloud job:
https://github.com/cosmos/interchain-security/pull/1443/checks?check_run_id=19046483938
https://github.com/cosmos/interchain-security/pull/1339/checks?check_run_id=19055224418
vs the details page of the repo analysis job
https://github.com/cosmos/interchain-security/actions/runs/7006381975/job/19058262889?pr=1449

Reply:
The difference is the if condition `if: ${{ env.GIT_DIFF }}`, i.e., 
```
      # before
      - name: SonarCloud Scan
        uses: SonarSource/sonarcloud-github-action@master
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # Needed to get PR information, if any
          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
        if: env.SONAR_TOKEN != null
```
```
     # now
     - name: sonarcloud
        if: ${{ env.GIT_DIFF && !github.event.pull_request.draft }}
        uses: SonarSource/sonarcloud-github-action@master
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
```
If you look at https://github.com/cosmos/interchain-security/actions/runs/7006381975/job/19058262889?pr=1449, you'll see that sonarcloud is not executed. That's because of the following lines
```
     - uses: technote-space/get-diff-action@v6.1.2
        id: git_diff
        with:
          PATTERNS: |
            **/*.go
            go.mod
            go.sum
            **/go.mod
            **/go.sum
```

Reply:
Does something else stop it from running here? https://github.com/cosmos/interchain-security/actions/runs/7008218190/job/19064178033?pr=1456
I am changing lines in a .go file, and afaict it is also not excluded in the sonarcloud properties